### PR TITLE
Custom Target Page

### PR DESCRIPTION
**Pull request by jlpoltrack**
From: https://github.com/olliw42/mLRS-docu/issues/22

@olliw42 Here's the direction that I'm starting with - would be good to know if there's any feedback.

### CODE REVIEW

**Comment by olliw42** [comment]
@jlpoltrack 

that's a great first attempt

I like the intro, and having examples, what I would miss is a more general description of how to handle custom boards

This is not exactly what I would suggest, but to give an idea let me just draft up something (the chapter should be the first I guess)

## Handling custom boards

There are a number of ways of how to create firmware for a custom board. The simplest would be to identify an existing board which is close to the custom board and to simply modify the entries in this boards hal file. The most advanced would be to create a new subproject/new target like you find them for the existing boards. This however requires additional skills and is more elaborious. Here we suggest another procedure, which is nearly as simple as the simplest but separates out the changes better (making it e.g. easier to update the code base).

1. Identify a board/target which is close to your new board. Example: The `rx-diy-board01-f103cb` board/target may be it.
2. Copy the existing board's hal file and rename it to what you want it to be. Example: Copy `/Common/hal/rx-hal-diy-board01-f103cb.h` to `/Common/hal/rx-hal-diy-customboard-f103cb.h`.
3. In `/Common/hal/hal.h` go to the area where the existing board's hal file is included, comment out the include, and add a line which inlcudes your hal file. Example: Find the line with `#include "rx-hal-diy-board01-f103cb.h"`, replace it with `\\#include "rx-hal-diy-board01-f103cb.h"` and insert a line `#include "rx-hal-diy-customboard-f103cb.h"`.
4. In `/Common/hal/device_conf.h` go to the area where the existing board is defined, comment out the define of the device name, and add your own define changing the device name to what you want it to be. Example: Find the line with `#ifdef RX_DIY_BOARD01_F103CB`, replace the subsequent line with `  \\#define DEVICE_NAME "DIY DualSX F103CB"` and insert a line `  #define DEVICE_NAME "My Own Great Board"` (the device name can be 20 characters max).

Questions:
Is this really what we would want to suggest? Would you have better recommendations?
The talk of "board" vs "target" may be confusing. Do we need to explain/define the difference? When should we use "board", when "target"?
????

**Comment by olliw42** [comment]
ah, that's a clear definition of board and target, I like it :)

**Comment by jlpoltrack** [comment]
Reworked the page to follow the steps:

1. Definitions
2. Assumptions
3. Creating a target
4. Modifying a target
5. Examples (still need to add a couple more)

File: docs/CUSTOM_BOARDS_TARGETS.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+## Creating a target for a custom board
+
+There are a number of ways of how to create a target for a custom board, for example:
+- A simple way would be to identify an existing board which is close to your custom board and modify the entries in this boards hal file.
+- A more advanced way would be to create a new target as you find them for the existing boards. This requires more work and additional skills.
+
+The suggested procedure is simple and makes it easier to update the code base by seperating out changes:
+ 
+1. Identify a target which is close to your new board. Example: The `rx-diy-board01-f103cb` target may be it.
+    1. Note: the only condition when picking a target is that you will need to choose a target which uses the same MCU as your custom board, in this example it is assumed that you're using the STM32F103CBXX on your custom board.
```
Comment (olliw42) [inline docs/CUSTOM_BOARDS_TARGETS.md:26]: maybe use - instead of 1. (dito below)
maybe wirte Note: The ...

File: docs/CUSTOM_BOARDS_TARGETS.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+2. Copy the existing target's hal file and rename it to what you want it to be. Example: Copy `/Common/hal/rx-hal-diy-board01-f103cb.h` to `/Common/hal/rx-hal-diy-customboard-f103cb.h`.
+3. In `/Common/hal/hal.h` go to the area where the existing target's hal file is included, comment out the include, and add a line which inlcudes your hal file. Example: Find the line with `#include "rx-hal-diy-board01-f103cb.h"`, replace it with `\\#include "rx-hal-diy-board01-f103cb.h"` and insert a line `#include "rx-hal-diy-customboard-f103cb.h"`.
+4. In `/Common/hal/device_conf.h` go to the area where the existing target is defined, comment out the define of the device name, and add your own define changing the device name to what you want it to be. Example: Find the line with `#ifdef RX_DIY_BOARD01_F103CB`, replace the subsequent line with `\\#define DEVICE_NAME "DIY DualSX F103CB"` and insert a line `#define DEVICE_NAME "My Own Great Board"`
+    1. Note: the device name can be 20 characters max.
```
Comment (olliw42) [inline docs/CUSTOM_BOARDS_TARGETS.md:30]: - instead of 1. ?

File: docs/CUSTOM_BOARDS_TARGETS.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+## Modifying a target
+
+The codebase has two locations that defines the features and capabilites of a target:
+1. Device Conf
+    1. Defines the device name, type (Tx/Rx), Sx chipset, frequencies
+    2. Location: mLRS/mLRS/Common/hal/device_conf.h
+        
+2. Target Hal
+    1. Defines the features, UART assignment and pin mapping
+    2. Location: mLRS/mLRS/Common/hal/TARGET.h
```
Comment (olliw42) [inline docs/CUSTOM_BOARDS_TARGETS.md:42]: TARGET is a bit strange here, maybe my-target. Maybe add an example, like inteh above.

**Comment by olliw42** [comment]
from the fact that you included the one chapter, may I assume that you think that this is indeed a recommendable approach?

few more examples could be indeed good, but I think it could be LGTM already now

File: docs/CUSTOM_BOARDS_TARGETS.md (hunk 1 above)
Comment (jlpoltrack) [reply docs/CUSTOM_BOARDS_TARGETS.md:26]: Done

File: docs/CUSTOM_BOARDS_TARGETS.md (hunk 2 above)
Comment (jlpoltrack) [reply docs/CUSTOM_BOARDS_TARGETS.md:30]: Done

File: docs/CUSTOM_BOARDS_TARGETS.md (hunk 3 above)
Comment (jlpoltrack) [reply docs/CUSTOM_BOARDS_TARGETS.md:42]: Updated to 'MY-CUSTOM-TARGET.h'

**Comment by jlpoltrack** [comment]
> from the fact that you included the one chapter, may I assume that you think that this is indeed a recommendable approach?

Yes, I think it makes sense to have all this information in one place.

Further updates:
- Completed the examples section
- Updated all of the file paths to use code syntax

Will merge and can incorporate any feedback in the future.  Enjoy your vacation.

**Pull request merged**